autoload/clap/cache.vim

```diff
@@ -0,0 +1,49 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: Cache API for clap.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+let s:path_separator = has('win32') ? '\' : '/'
+let s:clap_cache_dir = get(g:, 'clap_cache_dir', '')
+
+function! clap#cache#folder() abort
+  if empty(s:clap_cache_dir)
+    let user_home = $HOME
+    let user_cache = has('nvim') ? stdpath('cache') :
+          \ exists('$XDG_CACHE_HOME') ? $XDG_CACHE_HOME : 
+          \ user_home . s:path_separator . '.cache'
+    let s:clap_cache_dir = user_cache . s:path_separator . 'clap'
```

Comment:
Don't use multiple `expr1` in one single inline, it's not good for the readability.

Reply:
Done!